src/crepe/api/Camera.h

```diff
@@ -20,33 +21,31 @@ class Camera : public Component {
 	 * \param id Unique identifier for the camera component.
 	 * \param bg_color Background color for the camera view.
 	 */
-	Camera(game_object_id_t id, const Color & bg_color);
+	Camera(game_object_id_t id, const Color & bg_color, const ivec2 & screen,
+		   const ivec2 & viewport, const double & zoom, const vec2 & offset = {0, 0});
 	~Camera(); // dbg_trace only
 
 public:
 	//! Background color of the camera view.
-	Color bg_color;
+	const Color bg_color;
 
-	//! Aspect ratio height for the camera.
-	double aspect_height = 480;
+	//! offset postion from the game object transform component
+	vec2 offset;
 
-	//! Aspect ratio width for the camera.
-	double aspect_width = 640;
+	//! screen the display size in pixels ( output resolution )
+	const ivec2 screen;
 
-	//! X-coordinate of the camera position.
-	double x = 0.0;
-
-	//! Y-coordinate of the camera position.
-	double y = 0.0;
+	//! viewport is the area of the world visible through the camera (in world units)
+	const ivec2 viewport;
```

Comment:
Hoe kunnen we een `ivec2` hebben voor world units?

Reply:
De game units zijn eenheidloos dus waarom vind je dat `vec2` *te* precies is?

Reply:
ik snap niet helemaal at je bedoelt. dit wat de window in de game dus niks met hoe groot de wereld is? en waarom zou je een viewport grootte van `18.5,255.5' als sdl dat toch al niet support